packages/blade/src/components/Collapsible/CollapsibleLink.tsx

```diff
@@ -0,0 +1,86 @@
+import type { ReactElement } from 'react';
+import { useCallback } from 'react';
+import styled from 'styled-components';
+import { useCollapsibleContext } from './CollapsibleContext';
+import type { CollapsibleProps } from './Collapsible';
+import type { IconComponent } from '~components/Icons';
+import { ChevronDownIcon } from '~components/Icons';
+import type { LinkProps } from '~components/Link';
+import { Link } from '~components/Link';
+import { BaseBox } from '~components/Box/BaseBox';
+import { castWebType, makeMotionTime } from '~utils';
+
+type CollapsibleLinkProps = Pick<
+  LinkProps,
+  'size' | 'isDisabled' | 'testID' | 'accessibilityLabel' | 'children'
+>;
+
+type StyledCollapsibleLinkIconProps = {
+  isExpanded: boolean;
+  direction: CollapsibleProps['direction'];
+};
+
+// TODO: refactor common transition properties
+const StyledCollapsibleLinkIcon = styled(BaseBox)<StyledCollapsibleLinkIconProps>((props) => {
+  const { isExpanded, direction, theme } = props;
+  let transformExpanded, transformCollapsed;
+  if (direction === 'bottom') {
+    transformExpanded = 'rotate(0.5turn)';
+    transformCollapsed = undefined;
+  } else {
+    transformExpanded = undefined;
+    transformCollapsed = 'rotate(0.5turn)';
+  }
+
+  return {
+    display: 'flex',
+    alignItems: 'center',
+    transform: isExpanded ? transformExpanded : transformCollapsed,
```

Comment:
🤔 why do we need two variables? can't we do isExpanded 90deg or 0deg?

Reply:
We need to derive these based on the `direction` prop so kept two variables which derive these based on `direction` and then here I set them based on `isExpanded`

Reply:
still didn't understand 😅 ?

Reply:
The orientation of chevron icon inverts based on the direction collapsible expands in. `transformExpanded` and `transformCollapsed` therefore need to swap their corresponding expanded and collapsed values.

https://github.com/razorpay/blade/assets/6682655/f33a787c-e62d-4f7a-9a86-aa3ab8da29b0

I've also added this as a comment in #1296